### PR TITLE
feat(clojure): add implode and explode

### PR DESCRIPTION
**Pull request by elianiva**

### CODE REVIEW

**Comment by elianiva** [comment]
brb benerin CI

File: clojure/src/pehape/string.clj [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+   the last -limit elements is returned. If limit is 0 then an array with
+   one element is returned.
+   Returns:
+   - An array of strings."
+  {:added "0.0.1"}
+  ([separator content]
+   (cond
+     (or (nil? separator)
+         (= separator "")) (throw (IllegalArgumentException. "separator cannot be empty"))
+     (nil? content) (throw (IllegalArgumentException. "content cannot be empty"))
```
Comment (ronnygunawan) [inline clojure/src/pehape/string.clj:23]:
```suggestion
     (nil? content) (throw (IllegalArgumentException. "content cannot be nil"))
```

File: clojure/src/pehape/string.clj [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+   elements is returned. If limit is less than 0 then an array excluding
+   the last -limit elements is returned. If limit is 0 then an array with
+   one element is returned.
+   Returns:
+   - An array of strings."
+  {:added "0.0.1"}
+  ([separator content]
+   (cond
+     (or (nil? separator)
+         (= separator "")) (throw (IllegalArgumentException. "separator cannot be empty"))
```
Comment (ronnygunawan) [inline clojure/src/pehape/string.clj:22]:
```suggestion
         (= separator "")) (throw (IllegalArgumentException. "separator cannot be nil or empty"))
```

File: clojure/src/pehape/string.clj [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+     (nil? content) (throw (IllegalArgumentException. "content cannot be empty"))
+     :else (str/split content (re-pattern separator))))
+  ([separator content limit]
+   (cond
+     (or (nil? separator)
+         (= separator "")) (throw (IllegalArgumentException. "separator cannot be empty"))
```
Comment (ronnygunawan) [inline clojure/src/pehape/string.clj:28]:
```suggestion
         (= separator "")) (throw (IllegalArgumentException. "separator cannot be nil or empty"))
```

File: clojure/src/pehape/string.clj [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+     (nil? content) (throw (IllegalArgumentException. "content cannot be empty"))
```
Comment (ronnygunawan) [inline clojure/src/pehape/string.clj:29]:
```suggestion
     (nil? content) (throw (IllegalArgumentException. "content cannot be nil"))
```

**Review by ronnygunawan — CHANGES_REQUESTED** [review]
Everything else LGTM

File: clojure/test/pehape/string_test.clj [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    (def content "Hello  pehape  world")
+    (is (= (explode " " content) ["Hello" "" "pehape" "" "world"])))
+  (testing "Can Explode Space Separated String By Double Space"
+    #_{:clj-kondo/ignore [:inline-def]}
+    (def content "Hello pehape world")
+    (is (= (explode "  " content) ["Hello pehape world"])))
+  (testing "Explode Space Separated String By Empty String Throws Exception"
+    #_{:clj-kondo/ignore [:inline-def]}
+    (def content "Hello pehape world")
+    (is (thrown-with-msg? IllegalArgumentException #"separator cannot be nil or empty" (explode "" content)))))
```
Comment (ronnygunawan) [inline clojure/test/pehape/string_test.clj]: Missing tests:
- [ ] `(explode nil content)`
- [ ] `(explode " " nil)`
- [ ] `(explode nil content 1)`
- [ ] `(explode " " nil 1)`

**Review by ronnygunawan — CHANGES_REQUESTED** [review]
Buat nambah coverage

**Review by ronnygunawan — APPROVED** [review]
Mantap LGTM

**Pull request merged**